### DISCUSSION TITLE
ESP32S3-4827S043 Quick reference

### DISCUSSION THREAD

**Discussion by ColinGarySmith** [moononournation/Arduino_GFX#587] [Show and tell]
Picked up one of these boards affectionally known as CYB (chinese yellow board) displays from aliexpress. (also available on ebay).
https://www.aliexpress.com/item/1005004788147691.html

it was at first very hard to get running, but after some research the following tips and tricks will help you get this board working (and fast).

1. ESP 32 settings. Use the ESP32S3 Dev Module as the board in Arduino IDE or VScode.

2. Board settings:
Board: “ESP323 Dev Module”
USB CDC On Boot: “Disabled”
CPU Frequency: “240MHz (WiFi)”
Core Debug Level: “None”
USB DFU On Boot: “Disabled”
Erase All Flash Before Sketch Upload: “Disabled”
Events Run On: “Core 0”
Flash Mode: “QI0 80MHz”
Flash Size: “16MB (128Mb)”
JTAG Adapter: “Disabled”
Arduino Runs On: “Core 1”
USB Firmware MSC On Boot: “Disabled”
Partition Scheme: “8M with spiffs (3MB APP/1.5MB SPIFFS)”
PSRAM: “Disabled”
Upload Mode: “UARTO / Hardware CDC”
Upload Speed: “921600”
USB Mode: “Hardware CDC and JAG”

3. Definition to get the display working:

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
    45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
    5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
    8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
    0 /* hsync_polarity */, 1 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
    0 /* vsync_polarity */, 3 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 12 /* vsync_back_porch */,
    1 /* pclk_active_neg */, 8000000 /* prefer_speed */);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    480 /* width */, 272 /* height */, rgbpanel, 0 /* rotation */, false /* auto_flush */);

not sure why but the autoflash being true or false didn't seem to change much, however setting it to false gained some speed.
Speed set to 8000000. Faster this goes, the quicker the refresh happens, but you lose performance with the drawing operations. 

4. Touch screen (this is for the resistive touch screen) these settings are in touch.h from the example files.
#define TOUCH_XPT2046
#define TOUCH_XPT2046_SCK 12
#define TOUCH_XPT2046_MISO 13
#define TOUCH_XPT2046_MOSI 11
#define TOUCH_XPT2046_CS 38
#define TOUCH_XPT2046_INT 18
#define TOUCH_XPT2046_ROTATION 0
#define TOUCH_XPT2046_SAMPLES 50

5. Download the libs mentioned in touch.h to get the touch working. 

6.  Backlight pin is pin 2. It IS dimmable. 
pinMode(2, OUTPUT);
analogWrite(2, brightness);

The usuable range is from about 10 to 130. non-linear. 

If i have any other tips i will put them here.

**Comment by akomakom** [comment]
@ColinGarySmith  This is very helpful.  

Do you have any examples of migrating from the bundled code (zip files on the product page, meant for old versions of Arduino GFX Library, LVGL and ESP32 2.x) to modern versions of these?   What versions of these three were you able to use?

Thanks!

I am unable to get this code to work with the latest ESP32 boards (v3.1.3) + latest Arduino GFX Library (v1.5.3).  Here is the code from above plus some graphics testing code:

```c++

#include <Arduino_GFX_Library.h>

#define TFT_BL 2

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
0 /* hsync_polarity */, 1 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
0 /* vsync_polarity */, 3 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 12 /* vsync_back_porch */,
1 /* pclk_active_neg */, 8000000 /* prefer_speed */);

Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
480 /* width */, 272 /* height */, rgbpanel, 0 /* rotation */, false /* auto_flush */);

/*******************************************************************************
 * End of Arduino_GFX setting
 ******************************************************************************/

void setup(void)
{

  Serial.begin(115200);
  // Serial.setDebugOutput(true);
  // while(!Serial);
  Serial.println("Arduino_GFX Hello World example");

  // Init Display
  if (!gfx->begin())
  {
    Serial.println("gfx->begin() failed!");
  }
  gfx->fillScreen(RGB565_BLACK);

#ifdef TFT_BL
  pinMode(TFT_BL, OUTPUT);
  digitalWrite(TFT_BL, HIGH);
#endif

  gfx->setCursor(10, 10);
  gfx->setTextColor(RGB565_RED);
  gfx->println("Hello World!");

  delay(5000); // 5 seconds
}

void loop()
{
  gfx->setCursor(random(gfx->width()), random(gfx->height()));
  gfx->setTextColor(random(0xffff), random(0xffff));
  gfx->setTextSize(random(6) /* x scale */, random(6) /* y scale */, random(2) /* pixel_margin */);
  gfx->println("Hello World!");

  delay(1000); // 1 second
}
```

At runtime, I see:

```
============ Before Setup End ============
Arduino_GFX Hello World example
E (361) lcd_panel.rgb: esp_lcd_new_rgb_panel(290): fb size must be even multiple of bounce buffer size
ESP_ERROR_CHECK failed: esp_err_t 0x102 (ESP_ERR_INVALID_ARG) at 0x420034fc
file: "/home/akom/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_ESP32RGBPanel.cpp" line 134
func: uint16_t* Arduino_ESP32RGBPanel::getFrameBuffer(int16_t, int16_t)
expression: esp_lcd_new_rgb_panel(&panel_config, &_panel_handle)

abort() was called at PC 0x4037d137 on core 1


Backtrace: 0x4037664e:0x3fca4000 0x4037d141:0x3fca4020 0x40383b9d:0x3fca4040 0x4037d137:0x3fca40c0 0x420034fc:0x3fca40f0 0x420035ea:0x3fca41c0 0x4200236d:0x3fca41e0 0x420088df:0x3fca4210 0x4037dd5a:0x3fca4230




ELF file SHA256: 7ea660a2d

Rebooting...
```

Opened #628 about this

**Reply by ThomDyson** [reply]
Shouldn't CYD's be using the SPI interface?  Check this file for sample configs. Find your display driver. https://github.com/moononournation/Arduino_GFX/blob/master/examples/PDQgraphicstest/Arduino_GFX_dev_device.h

**Reply by akomakom** [reply]
@ThomDyson my understanding is that this model has a parallel interface.  It's working with v1.5.0, and the header file you linked supports the way I'm using it.

**Reply by ColinGarySmith** [reply]
> Shouldn't CYD's be using the SPI interface? Check this file for sample configs. Find your display driver. https://github.com/moononournation/Arduino_GFX/blob/master/examples/PDQgraphicstest/Arduino_GFX_dev_device.h

It depends entirely on the board. I have 2 here, a smaller screen but larger resolution one that runs on SPI (and as you can imagine, slow screen refreshes) and a large screen board, but with lower res that uses the RGB656 interface that is incredibly fast.  The spi one could probably be tuned to be a "bit" faster, but it will never be as fast as the direct drive RGB656.

**Comment by akomakom** [comment]
Additionally, the partition scheme option suggested in the original post didn't work for me on this board:

```
E (31) flash_parts: partition 3 invalid - offset 0x340000 size 0x330000 exceeds flash chip size 0x400000
E (32) boot: Failed to verify partition table
E (33) boot: load partition table error!
```
Default 4MB with spiffs (first option) works fine, and also this works for arduino-cli:
`arduino-cli compile --fqbn esp32:esp32:esp32s3 --board-options "PSRAM=opi,FlashSize=16M,DebugLevel=warn"`

**Reply by ColinGarySmith** [reply]
> Additionally, the partition scheme option suggested in the original post didn't work for me on this board:
> 
> ```
> E (31) flash_parts: partition 3 invalid - offset 0x340000 size 0x330000 exceeds flash chip size 0x400000
> E (32) boot: Failed to verify partition table
> E (33) boot: load partition table error!
> ```
> 
> Default 4MB with spiffs (first option) works fine, and also this works for arduino-cli: `arduino-cli compile --fqbn esp32:esp32:esp32s3 --board-options "PSRAM=opi,FlashSize=16M,DebugLevel=warn"`

The flash option seems to be a hit and miss im afraid, you can try and changing the flash size in the programmer to find one that works.

All these CYBs don't seem to follow any kind of standard or "understanding" between models and its a hit and miss to get one working easily i have found. The hardest bit however is working out what display interface is being used. I ended up actually getting the circuit diagram for mine of the seller, which gave me a much greater understanding of how its all connected, and more importantly what display and display chip is being used. Once you have that info its a bit easier.

**Comment by DragonSlayerSE** [comment]
i have this ESP32-4827S043R display.
<img width="846" height="569" alt="image" src="https://github.com/user-attachments/assets/81a53f27-c332-4550-bdf8-65f2ab603c9c" />
The highest version of this library i can use is v1.6.0... higher version compile and upload, but then crashes. i have posted issues, but creator assumes i have somthing wrong in my config.. what could i be missing?
what do i need to change in Arduino IDE 2.3.6 , when i update library beyond v1.6.0?
<img width="472" height="682" alt="image" src="https://github.com/user-attachments/assets/b54cef93-3f52-4a9d-ad2c-875ed8040538" />
i am not a programmer and new to Arduino IDE ... in VS Code i could use this platformIO.ini content: (when importing .ino project)
*****************

[env:esp32_4827S043R]
platform = espressif32
board = esp32-s3-devkitc-1
framework = arduino
build_flags = -DBOARD_HAS_PSRAM
board_build.arduino.memory_type = qio_opi
board_build.f_flash = 80000000L
board_build.flash_mode = qio
; 16MB
board_upload.flash_size = 16MB
lib_extra_dirs = c:\Users\kired\OneDrive\Dokument\Arduino\libraries
monitor_speed = 115200

***********************
same here i can not get past 1.6.0 .. but in Platformio i get compile errors 

sample code:
***********************
#include <Arduino_GFX_Library.h>  // By Moon On Our Nation
#include <XPT2046_Touchscreen.h>  // By Paul Stoffregen
#include <SPI.h>

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
    45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
    5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
    8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */,
    0 /* hsync_polarity */, 1 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 43 /* hsync_back_porch */,
    0 /* vsync_polarity */, 3 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 12 /* vsync_back_porch */,
    1 /* pclk_active_neg */, 9000000 /* prefer_speed */);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    480 /* width */, 272 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */);


#define GFX_BL DF_GFX_BL // default backlight pin, you may replace DF_GFX_BL to actual backlight pin
#define TFT_BL 2

// Setup touch
#define TOUCH_XPT2046
#define TOUCH_XPT2046_SCK 12
#define TOUCH_XPT2046_MISO 13
#define TOUCH_XPT2046_MOSI 11
#define TOUCH_XPT2046_CS 38
#define TOUCH_XPT2046_INT 18
#define TOUCH_XPT2046_ROTATION 1
#define TOUCH_MAP_X1 0
#define TOUCH_MAP_X2 0
#define TOUCH_MAP_Y1 0
#define TOUCH_MAP_Y2 0

XPT2046_Touchscreen ts(TOUCH_XPT2046_CS, TOUCH_XPT2046_INT);

void setup(void) {
  Serial.begin(115200);
  gfx->begin();
  gfx->fillScreen(RGB565_BLACK);

#ifdef TFT_BL
  pinMode(TFT_BL, OUTPUT);
  digitalWrite(TFT_BL, HIGH);
#endif

  gfx->setCursor(10, 10);
  gfx->setTextColor(RGB565_RED);
  gfx->println("Hello World!");

  delay(1000);  // 1 seconds

  SPI.begin(TOUCH_XPT2046_SCK, TOUCH_XPT2046_MISO, TOUCH_XPT2046_MOSI, TOUCH_XPT2046_CS);
  ts.begin();
  ts.setRotation(TOUCH_XPT2046_ROTATION);

  gfx->fillScreen(RGB565_BLACK);
  gfx->setCursor(0, 0);
  gfx->print("ESP32-4827S043R touch test: v1.60");
  gfx->setTextColor(RGB565_GREEN);
  Serial.println(digitalRead(-1));
}

void loop() {
  if (ts.tirqTouched()) {
    if (ts.touched()) {
      // Get touched point
      TS_Point p = ts.getPoint();
      // Print touch in Serial Monitor
      Serial.print("Touch: ");
      Serial.print("  X: ");
      Serial.print(p.x);
      Serial.print("  Y: ");
      Serial.print(p.y);
      Serial.print("  Pressure: ");
      Serial.println(p.z);

      // Print touch on screen
      gfx->setCursor(0, 20);
      gfx->println("X: " + String(p.x) + "   Y: " + String(p.y) + "   Pressure: " + String(p.z));

      delay(100);
      gfx->fillRect(0, 20, 220, 40, RGB565_BLACK);
      gfx->fillRect(0, 300, 170, 20, RGB565_BLACK);
      gfx->setCursor(0, 310);
    }
  }
  delay(50);
}

***********************


Could these could this be the problem?
<img width="842" height="364" alt="image" src="https://github.com/user-attachments/assets/1dfff121-3d8d-405e-b39d-ff8fde37a41f" />

i dont know what values to put in after "prefer_speed" (or if i have to)

the error in serial monitor after compiling and upload v1.6.3:

ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40378fd6
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce2820,len:0x116c
load:0x403c8700,len:0xc2c
load:0x403cb700,len:0x3108
entry 0x403c88b8
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.


Core  1 register dump:
PC      : 0x4200376f  PS      : 0x00060130  A0      : 0x82003241  A1      : 0x3fcebc30  
A2      : 0x3fc9d42c  A3      : 0x0000000a  A4      : 0x0000000b  A5      : 0x0000f800  
A6      : 0x4200372c  A7      : 0x00000010  A8      : 0x00000001  A9      : 0x0000000a  
A10     : 0xbcd62454  A11     : 0x00000000  A12     : 0x0000001f  A13     : 0x6f2aad56  
A14     : 0x6f2aad60  A15     : 0x00000020  SAR     : 0x0000001e  EXCCAUSE: 0x0000001d  
EXCVADDR: 0xbcd62454  LBEG    : 0x400556d5  LEND    : 0x400556e5  LCOUNT  : 0xfffffffc  

Backtrace: 0x4200376c:0x3fcebc30 0x4200323e:0x3fcebc50 0x42027c6d:0x3fcebc90 0x42028133:0x3fcebcb0 0x42004e77:0x3fcebcd0 0x42004ec9:0x3fcebcf0 0x420022db:0x3fcebd10 0x420092c1:0x3fcebd40 0x4037d225:0x3fcebd60




ELF file SHA256: 31576cd90

Rebooting...
